processing/src/test/java/org/apache/druid/java/util/metrics/HttpPostEmitterMonitorTest.java

```diff
@@ -0,0 +1,129 @@
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.java.util.metrics;
+
+import com.google.common.collect.ImmutableMap;
+import org.apache.druid.java.util.emitter.core.ConcurrentTimeCounter;
+import org.apache.druid.java.util.emitter.core.HttpPostEmitter;
+import org.apache.druid.java.util.emitter.service.ServiceEmitter;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.util.List;
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class HttpPostEmitterMonitorTest
+{
+  private HttpPostEmitter mockHttpPostEmitter;
+  private ServiceEmitter mockServiceEmitter;
+  private HttpPostEmitterMonitor monitor;
+
+  @BeforeEach
+  public void setUp()
+  {
+    mockHttpPostEmitter = mock(HttpPostEmitter.class);
+    mockServiceEmitter = mock(ServiceEmitter.class);
+    monitor = new HttpPostEmitterMonitor("testFeed", mockHttpPostEmitter,
+            ImmutableMap.of("dimensionKey", "dimensionValue"));
+  }
+
+  @Test
+  public void testDoMonitor()
+  {
+    when(mockHttpPostEmitter.getTotalEmittedEvents()).thenReturn(100L);
+    when(mockHttpPostEmitter.getTotalDroppedBuffers()).thenReturn(10);
+    when(mockHttpPostEmitter.getTotalAllocatedBuffers()).thenReturn(20);
+    when(mockHttpPostEmitter.getTotalFailedBuffers()).thenReturn(5);
+    when(mockHttpPostEmitter.getBatchFillingTimeCounter()).thenReturn(mock(ConcurrentTimeCounter.class));
+    when(mockHttpPostEmitter.getSuccessfulSendingTimeCounter()).thenReturn(mock(ConcurrentTimeCounter.class));
+    when(mockHttpPostEmitter.getFailedSendingTimeCounter()).thenReturn(mock(ConcurrentTimeCounter.class));
+    when(mockHttpPostEmitter.getEventsToEmit()).thenReturn(200L);
+    when(mockHttpPostEmitter.getLargeEventsToEmit()).thenReturn(75L);
+    when(mockHttpPostEmitter.getBuffersToEmit()).thenReturn(30);
+    when(mockHttpPostEmitter.getBuffersToReuse()).thenReturn(15);
+
+    final StubServiceEmitter stubServiceEmitter = new StubServiceEmitter("service", "host");
+
+    assertTrue(monitor.doMonitor(stubServiceEmitter));
+
+    Map<String, List<StubServiceEmitter.ServiceMetricEventSnapshot>> metricEvents = stubServiceEmitter.getMetricEvents();
+
+    final String emitter_successful_sending_max_time_ms = "emitter/successfulSending/maxTimeMs";
+    assertEquals(metricEvents.get(emitter_successful_sending_max_time_ms).get(0).getMetricEvent().getMetric(), emitter_successful_sending_max_time_ms);
+    assertEquals(metricEvents.get(emitter_successful_sending_max_time_ms).get(0).getMetricEvent().getValue(), 0);
```

Comment:
Can you create a method for this logic, as it's redundant for each metric? You can pass in the metric name, expected value, and `metricEvents` to the method. This would significantly clean up the logic and improve readability.

You'd then be able to replace these assertions with something like:
```java
assertMetricValue(metricEvents, "emitter/successfulSending/maxTimeMs", 0);
```

Reply:
Done. Pushed in the next commit.